packages/cli/src/commands/set-admin.ts

```diff
@@ -19,46 +43,111 @@ const register: (program: any) => any = (program) => program
   .description(description)
   .option('-f, --force', 'bypass a manual check')
   .withNetworkOptions()
+  .withNonInteractiveOption()
   .action(action);
 
-async function action(contractFullNameOrAddress: string, newAdmin: string, options: any): Promise<void | never> {
-  const { network, txParams } = await ConfigVariablesInitializer.initNetworkConfiguration(options);
+async function action(contractReference: string, newAdmin: string, options: any): Promise<void | never> {
+  const { force, network: networkInOpts, interactive  } = options;
+  const networkOpts = await promptForNetwork(options, () => getCommandProps());
+
+  const { network, txParams } = await ConfigVariablesInitializer.initNetworkConfiguration({ ...options, ...networkOpts });
   if (!await hasToMigrateProject(network)) process.exit(0);
 
-  let proxyAddress;
-  let contractAlias;
-  let packageName;
+  const { proxyReference, newAdmin: pickedNewAdmin } = await promptForProxies(contractReference, newAdmin, network, options);
+  const parsedContractReference = parseContractReference(proxyReference);
 
-  if (!contractFullNameOrAddress) throw Error('You have to specify at least a new admin address.');
+  if (!parsedContractReference) throw Error('You have to specify at least a new admin address.');
 
-  // we assume if newAdmin is empty it was specified as first argument
-  if (!newAdmin) {
-    newAdmin = contractFullNameOrAddress;
-    contractFullNameOrAddress = '';
-  }
-
-  if (!options.force) {
+  // has to be a standalone question from interactivity
+  // because it is security related and can't be disabled with interactivity set to false
+  if (!force) {
     const answers = await inquirer.prompt([
       {
         name: 'address',
         type: 'string',
         message: 'Please double check your address and type the last 4 characters of the new admin address. If you provide a wrong address, you will lose control over your contracts.',
       }
     ]);
-    if (answers.address !== newAdmin.slice(-4)) {
+    if (answers.address.toLowerCase() !== pickedNewAdmin.slice(-4).toLowerCase()) {
```

Comment:
```suggestion
    if (pickedNewAdmin && answers.address.toLowerCase() !== pickedNewAdmin.slice(-4).toLowerCase()) {
```
I believe that if the user do not specify the new admin address, and is using the `--no-interactive` flag, `zos` will error something like: `cannot read property slice of undefined`, as `newAdmin` will be `undefined`. You will get rid of this error if we only prompt this question when not specifying the `--no-interactive` though.

Reply:
Just run a test and it worked. Have you considered this code?
```
if (!newAdmin) {
    newAdmin = contractReference;
    contractReference = '';
  }
```

Reply:
Yeap, I've seen it. But if the user runs `zos set-admin --no-interactive --network development` It will throw that ugly error instead of gently erroring with a "Please provide a contract address/admin address/whatever`.

Reply:
So, true. Fixed it.